### ISSUE TITLE
Plugin Generator

### ISSUE THREAD

**Issue by repeatedly** [fluent/fluentd#1376]
For easy plugin development, generating plugin template is useful for users.

## Command

```
$ fluent-plugin-generate type name
```

Based on Embulk: http://www.embulk.org/docs/customization.html#id1

## Generated files

```
.
├── ChangeLog
├── Gemfile
├── README.md
├── Rakefile
├── fluent-plugin-xxx.gemspec
├── lib
│   └── fluent
│       └── plugin
│           └── out_xxx.rb
└── test
    └── test_out_xxx.rb
```

Generated plugin file is based on `type` parameter. If `type` is input, file is `in_xxx.rb`
If same file exist, command prefer existing file.

**Comment by tagomoris** [comment]
I have an idea to add a command to:
* load plugin .rb file
* get the list of configuration parameters (and its type, options and descriptions)
* get the list of plugin helper loaded
* generate a well-formatted Markdown section (for copy-and-paste to README) to describe configuration parameters and links to the page of available plugin helper features (i know it doesn't exist yet)

It's better to add a placeholder in default README.md (generated by template) to be replaced by the generated section above.

**Comment by okkez** [comment]
I will work for this issue if nobody has started development.

**Comment by repeatedly** [comment]
Fixed by #1427
